src/main/java/io/fabric8/jenkins/openshiftsync/BuildConfigWatcher.java

```diff
@@ -234,22 +226,38 @@ private void modifyJob(BuildConfig buildConfig) throws Exception {
     // no longer a Jenkins build so lets delete it if it exists
     deleteJob(buildConfig);
   }
-
-  private void deleteJob(final BuildConfig buildConfig) throws Exception {
-    final Job job = getJobFromBuildConfig(buildConfig);
-    if (job != null) {
-      ACL.impersonate(ACL.SYSTEM, new NotReallyRoleSensitiveCallable<Void, Exception>() {
-        @Override
-        public Void call() throws Exception {
-          try {
-            job.delete();
-          } finally {
-            removeJobWithBuildConfig(buildConfig);
-            Jenkins.getActiveInstance().rebuildDependencyGraphAsync();
+  
+  private void innerDeleteJob(final BuildConfig buildConfig) throws Exception {
+      final Job job = getJobFromBuildConfig(buildConfig);
+      if (job != null) {
+        ACL.impersonate(ACL.SYSTEM, new NotReallyRoleSensitiveCallable<Void, Exception>() {
+          @Override
+          public Void call() throws Exception {
+            try {
+              job.delete();
+            } finally {
+              removeJobWithBuildConfig(buildConfig);
+              Jenkins.getActiveInstance().rebuildDependencyGraphAsync();
+            }
+            return null;
           }
-          return null;
+        });
+      }
+      
+  }
+
+  private synchronized void deleteJob(final BuildConfig buildConfig) throws Exception {
+    String bcUid = buildConfig.getMetadata().getUid();
+    if (bcUid != null && bcUid.length() > 0) {
+        // employ intern to facilitate sync'ing on the same actual object
+        bcUid = bcUid.intern();
+        // for deletes we coordinate with builds ... cascading delete via `oc`
```

Comment:
got it, yeah i was reading it as this saying the code was going to cascade the deletes via oc, not that the code was handling delete events that were generated *by* an oc delete cascade.